src/edu/gatech/cs2340/trydent/Time.java

```diff
@@ -0,0 +1,103 @@
+package edu.gatech.cs2340.trydent;
+
+/**
+ * Static class to keep track of time.
+ * 
+ * All values returned by public methods are in seconds,
+ * unless stated otherwise.
+ * 
+ * @author Garrett Malmquist
+ */
+public class Time {
+
+	// All private variables are stored in milliseconds for accuracy.
+	private static volatile double gameTimeSinceStartup;
+	private static volatile long realTimeSinceStartup;
+	private static volatile long realLastFrame;
+	private static volatile long realStartTime;
+	private static volatile double gameTimePassed;
+	private static volatile long realTimePassed;
+	private static volatile double timeRate;
+	
+	/**
+	 * Game time since the game started.
+	 * @return
+	 */
+	public static double getTime() {
+		return gameTimeSinceStartup / 1000.0;
+	}
+	
+	/**
+	 * Game time elapsed since the previous frame.
+	 * @return
+	 */
+	public static double getTimePassed() {
+		return gameTimePassed / 1000.0;
+	}
+	
+	/**
+	 * Real time since the game started.
+	 * @return
+	 */
+	public static double getRealTimeSinceStartup() {
+		return realTimeSinceStartup / 1000.0;
+	}
+	
+	/**
+	 * Real time elapsed since the previous frame.
+	 * @return
+	 */
+	public static double getRealTimePassed() {
+		return realTimePassed / 1000.0;
+	}
+	
+	/**
+	 * Rate of game time relative to real-time. I.e, a 
+	 * time rate of 0.5 indicates that the game is running
+	 * at half-speed (in slow motion).
+	 * @return
+	 */
+	public static double getTimeRate() {
+		return timeRate;
+	}
+	
+	/**
+	 * Sets the rate of game time relative to real-time. I.e,
+	 * a time rate of 0.5 indicates that the game is running
+	 * at half-speed (in slow motion).
+	 * @param rate
+	 */
+	public static void setTimeRate(double rate) {
+		timeRate = rate;
+	}
+	
+	/**
+	 * Starts keeping track of time, called by the TrydentEngine.
+	 */
+	protected static void startTheDawnOfTime() {
+		realStartTime = System.currentTimeMillis();
```

Comment:
While this is totally reasonable, do we instead perhaps want to use System.nanoTime()? As far as I am aware, these are identical except that nanoTime may have a higher resolution on some machines.